packages/admin-app/src/components/ElectionSetupWizard/Steps/ManifestUploadStep.tsx

```diff
@@ -22,116 +21,100 @@ export const alert = (props: AlertProps) => (
 );
 
 const useStyles = makeStyles((theme) => ({
-    root: {
-        flexGrow: 1,
-        height: '100%',
-    },
     content: {
         paddingTop: theme.spacing(4),
         paddingBottom: theme.spacing(4),
         display: 'flex',
         flexDirection: 'column',
         alignItems: 'center',
     },
-    wrapper: {
-        margin: theme.spacing(1),
-        position: 'relative',
-    },
-    buttonProgress: {
-        position: 'absolute',
-        top: '50%',
-        left: '50%',
-        marginTop: -12,
-        marginLeft: -12,
+    error: {
+        color: 'red',
+        marginBottom: theme.spacing(2),
     },
 }));
 
 export interface ManifestUploadStepProps {
     onNext: () => void;
-    uploadManifest: (manifestFile: File) => Promise<boolean>;
+    onUploadManifest: (manifestRequest: ValidateManifestRequest) => void;
 }
 
-const ManifestUploadStep: React.FC<ManifestUploadStepProps> = ({ onNext, uploadManifest }) => {
+const ManifestUploadStep: React.FC<ManifestUploadStepProps> = ({ onNext, onUploadManifest }) => {
     const [uploading, setUploading] = useState(false);
-    const [error, setError] = useState(false);
+    const [error, setError] = useState<string>();
+    const intl = useIntl();
+
+    const setIntlError = (id: string) => {
+        const message = intl.formatMessage({
+            id,
+        });
+        setError(message);
+    };
 
     const handleClose = (_event?: React.SyntheticEvent | Event, reason?: SnackbarCloseReason) => {
         if (reason === 'clickaway') {
             return;
         }
 
-        setError(false);
+        setError(undefined);
     };
 
     const onFileUpload = async (e: React.ChangeEvent<HTMLInputElement>) => {
         if (e?.target?.files !== null) {
             const file = e.target.files[0];
             setUploading(true);
-            const success = await uploadManifest(file);
-            setUploading(false);
-            setError(!success);
-            if (success) onNext();
+            try {
+                const text = await file.text();
+                const json = JSON.parse(text);
+                if (!json.manifest) {
+                    throw new Error();
```

Comment:
Exact same pattern as the key upload PR, if that wasn't great this won't be either.  Happy to fix both if this isn't idiomatic.